cmake/HPX_AddModule.cmake

```diff
@@ -0,0 +1,166 @@
+  cmake_parse_arguments(${name} "${options}" "${one_value_args}" "${multi_value_args}" ${ARGN})
+
+  # Compatibility header off is not specified
+  if (NOT ${name}_COMPATIBILITY_HEADERS)
+      set(${name}_COMPATIBILITY_HEADERS OFF)
+  endif()
+
+  # Main directories of the module
+  set(SOURCE_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/src")
+  hpx_debug("Add module ${name}: SOURCE_ROOT: ${SOURCE_ROOT}")
+  set(HEADER_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/include")
+  hpx_debug("Add module ${name}: HEADER_ROOT: ${HEADER_ROOT}")
+  set(COMPAT_HEADER_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/include_compatibility")
+  hpx_debug("Add module ${name}: COMPAT_HEADER_ROOT: ${COMPAT_HEADER_ROOT}")
+
+  string(TOUPPER ${name} name_upper)
+
+  # HPX options
+
+  hpx_option(HPX_${name_upper}_WITH_TESTS
+    BOOL
+    "Build HPX ${name} module tests. (default: ON)"
+    ON ADVANCED
+    CATEGORY "Modules")
+
+  if(${name}_DEPRECATION_WARNINGS)
+    hpx_option(HPX_${name_upper}_WITH_DEPRECATION_WARNINGS
+      BOOL
+      "Enable warnings for deprecated facilities. (default: ${HPX_WITH_DEPRECATION_WARNINGS})"
+      ${HPX_WITH_DEPRECATION_WARNINGS} ADVANCED
+      CATEGORY "Modules")
+    if(HPX_${name}_WITH_DEPRECATION_WARNINGS)
+      hpx_add_config_define_namespace(
+          DEFINE HPX_${name_upper}_HAVE_DEPRECATION_WARNINGS
+          NAMESPACE ${name_upper})
+    endif()
+  endif()
+  
+  if(${name}_COMPATIBILITY_HEADERS)
+    # Added in 1.4.0
+    hpx_option(HPX_${name_upper}_WITH_COMPATIBILITY_HEADERS
+      BOOL
+      "Enable compatibility headers for old headers"
+      ON ADVANCED
+      CATEGORY "Modules")
+    if(HPX_${name_upper}_WITH_COMPATIBILITY_HEADERS)
+      hpx_add_config_define_namespace(
+          DEFINE HPX_${name_upper}_HAVE_COMPATIBILITY_HEADERS
+          NAMESPACE ${name_upper})
+    endif()
+  endif()
+
+  # Write full path for the sources files
+  include(HPX_CMakeUtils)
+  prepend(sources ${SOURCE_ROOT} ${${name}_SOURCES})
+  prepend(headers ${HEADER_ROOT} ${${name}_HEADERS})
+  if(HPX_${name_upper}_WITH_COMPATIBILITY_HEADERS)
+    prepend(compat_headers ${COMPAT_HEADER_ROOT} ${${name}_COMPAT_HEADERS})
+  endif()
+  # To make the global header file
+  set(tmp_headers ${${name}_HEADERS})
+
+  # Add a global include file that include all module headers
+  FILE(WRITE ${HEADER_ROOT}/${name}_module.hpp
+      "//  Copyright (c) 2019 The STE||AR GROUP\n"
+      "//\n"
+      "//  Distributed under the Boost Software License, Version 1.0. (See accompanying\n"
+      "//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)\n\n"
+  )
+  foreach(header_file ${tmp_headers})
+    FILE(APPEND ${HEADER_ROOT}/${name}_module.hpp
+      "#include <${header_file}>\n"
+    )
+    list(APPEND headers "${HEADER_ROOT}/${header_file}")
+  endforeach(header_file)
+
+  foreach(header_file ${headers})
+      hpx_debug(${header_file})
+  endforeach(header_file)
+
+  message(STATUS "${name}: Configuring")
```

Comment:
We recently switched to using `hpx_info()` here (see: https://github.com/STEllAR-GROUP/hpx/blob/master/libs/config/CMakeLists.txt#L16).

Reply:
sorry :/